### PR TITLE
fix(sql): fix sampling aligned to calendar timezone and ending with semicolon or followed by order-by/limit

### LINKED ISSUE #2390: ALIGN TO CALENDAR feels buggy
### Describe the bug

`ALIGN TO CALENDAR` has a weird behaviour affecting to clauses like `FILL` or `LIMIT`. Maybe to some others, but I am guessing the fix is the same

### To reproduce

Two problems here: `LIMIT` and `FILL`

## LIMIT 
If I go to the demo site and execute this query, it works
```
select  last(price), `symbol` from trades
SAMPLE BY 1h limit 100
```

If I now add any modifiers to `SAMPLE BY,` it breaks
```
select  last(price), `symbol` from trades
SAMPLE BY 1h ALIGN to calendar limit 100
```

However, if I add a `TIME ZONE` and an `OFFSET`, it works. Note that if I add *only* a `TIME ZONE` is not enough and requires and offset even if it is zero. It seems like a parsing issue. This works

```
select  last(price), `symbol` from trades
SAMPLE BY 1h  ALIGN TO CALENDAR TIME ZONE 'Europe/Berlin'
WITH OFFSET '00:00' limit 10
```

I am aware I can use subqueries to do the limit on the outer query, but developer experience here is not great as I would expect `SAMPLE BY` and `LIMIT` either to work together or not, but not something in between

## FILL

This query works
```
select  last(price), `symbol` from trades
SAMPLE BY 1h FILL(PREV) ALIGN TO CALENDAR 
```

This query doesn't
```
select  last(price), `symbol` from trades
SAMPLE BY 1h  ALIGN TO CALENDAR  FILL(PREV)
```

No matter if I add time zone and offset. It still fails. It says `FILL` token is unexpected. Unless I am reading [the diagram at the docs wrong](https://questdb.io/docs/reference/sql/sample-by/), `FILL` can be used after a `SAMPLE BY` query

### Expected Behavior

`LIMIT` should work with `SAMPLE BY` or not, but consistently

FILL should be accepted after `ALIGN` clause according to the docs. Maybe this is just a docs issue

`TIME ZONE `and `WITH OFFSET` should be optional according to the docs, and according to the diagram at the website it seems `WITH OFFSET `shouldn't be accepted after `TIME ZONE`, but that's probably a fix for the documentation itself

### Environment

```markdown
- **QuestDB version**: 6.4.3
- **OS**: Linux
- **Browser**: Chrome
```

### Additional context

This is the architecture diagram [at the doc site](https://questdb.io/docs/reference/sql/sample-by/)

image

### PR DESCRIPTION
**Pull request by jerrinot**
fixes #2390
this is a left-over from #2013

### CODE REVIEW

**Comment by questdb-butler** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please update the PR title to match this format:
`type(subType): description`

Where `type` is one of:
`feat`, `fix`, `chore`, `docs`, `style`, `refactor`, `perf`, `test`, `build`, `ci`, `chore`, `revert`

And: `subType` is one of:
`build`, `sql`, `log`, `mig`, `core`, `ilp`, `pgwire`, `http`, `conf`

For Example:

```
perf(sql): improve pattern matching performance for SELECT sub-queries
```
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3d59ab0d16df2d4373acc5628b15b94f5c827bff
</p>

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|1|1|100.00%|

**Review by bziobrowski — DISMISSED** [review]
Looks fine !

**Comment by jerrinot** [comment]
converting to a draft as it's not fully fixing the problem

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

File: core/src/test/java/io/questdb/griffin/engine/groupby/SampleByTest.java [hunk 1]
```diff
@@ -156,6 +156,39 @@ public void testGeohashInterpolated() throws Exception {
         );
     }
 
+    @Test
+    public void testSampleByAlignedToCalendarWithTimezoneEndingWithSemicolon() throws Exception {
+        assertQuery("k\tcount\n" +
+                        "1970-01-03T00:00:00.000000Z\t6\n" +
+                        "1970-01-03T06:00:00.000000Z\t6\n" +
+                        "1970-01-03T12:00:00.000000Z\t6\n" +
+                        "1970-01-03T18:00:00.000000Z\t2\n",
```
Comment (bziobrowski) [inline core/src/test/java/io/questdb/griffin/engine/groupby/SampleByTest.java:165]: Both tests would be a tiny bit more readable if "create table" was put in next line .

File: core/src/main/java/io/questdb/griffin/SqlParser.java [hunk 2]
```diff
@@ -1238,12 +1238,8 @@ private void parseFromClause(GenericLexer lexer, QueryModel model, QueryModel ma
                             model.setSampleByTimezoneName(expectExpr(lexer));
```
Comment (bziobrowski) [inline core/src/main/java/io/questdb/griffin/SqlParser.java:1238]: This piece does't seem to be consistent with sql diagram at https://questdb.io/docs/reference/sql/sample-by/
According to diagram _with offset_ is not allowed after _time zone 'X'_
```

**Comment by jerrinot** [comment]
fixes also https://github.com/questdb/questdb/issues/2285

File: core/src/main/java/io/questdb/griffin/SqlParser.java (hunk 2 above)
Comment (jerrinot) [reply core/src/main/java/io/questdb/griffin/SqlParser.java:1238]: for the record: I discussed this with Bolek as I had noticed the same. The grammar in docs makes it seem that it's either `with offset` (x)or `time zone`, but the impl allowed both in one query. not only the impl allows it, but there is even a test for this behaviour. 

I think it's quite useful, as it allows you to select a specific timezone and then apply an offset on top of that. that might be useful, as offset gives you finer granularity. so I think the docs should be updated, not the impl. I created https://github.com/questdb/questdb.io/issues/1115 to track this.

File: core/src/test/java/io/questdb/griffin/engine/groupby/SampleByTest.java (hunk 1 above)
Comment (jerrinot) [reply core/src/test/java/io/questdb/griffin/engine/groupby/SampleByTest.java:165]: agree and done!

**Comment by jerrinot** [comment]
@bziobrowski thank you for the review! I pushed the formatting change and create a docs issue to track the docs issue as discussed. does the PR look good to you now?

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Pull request merged**